osu.Game/Graphics/OsuColour.cs

```diff
@@ -63,6 +64,24 @@ public Color4 ForDifficultyRating(DifficultyRating difficulty, bool useLighterCo
             }
         }
 
+        public Color4 ForOverlayElement(OverlayColourScheme colourScheme, float saturation, float lightness, float opacity = 1) => Color4.FromHsl(new Vector4(getBaseHue(colourScheme), saturation, lightness, opacity));
+
+        private static float getBaseHue(OverlayColourScheme colourScheme)
+        {
+            var hue = colourScheme switch
+            {
+                OverlayColourScheme.Red => 0,
+                OverlayColourScheme.Pink => 333,
+                OverlayColourScheme.Orange => 46,
+                OverlayColourScheme.Green => 115,
+                OverlayColourScheme.Purple => 255,
+                OverlayColourScheme.Blue => 200,
+                _ => 0,
```

Comment:
Out-of-range values should throw instead of quietly falling back to `Red`.

Reply:
Is it even possible to set out-of-range value here?

Reply:
yes, by bugs.

Reply:
Enums are just really integer-type aliases, so you can forcibly cast any integer value and it'll get silently accepted but not get a nice printable name. Throwing also makes it easy to spot errors if somebody forgets to modify this switch, but adds a new enum value and uses it elsewhere.

Reply:
Oh, I see